src/typescript/azure-openapi-validator/rules/XmsEnumNameUnique.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from "../rule";
+export const XmsEnumNameUnique: string = "XmsEnumNameUnique";
+import { nodes } from "jsonpath";
+
+var enumList: string[] = []
+
+rules.push({
+  id: "R4005",
+  name: XmsEnumNameUnique,
+  severity: "warning",
+  category: "ARMViolation",
```

Comment:
It's not just ARM but everything autorest related

Reply:
for this rule ,  it  just impact code generation

Reply:
It impacts code generation cross-context, not only for ARM